### PR TITLE
comparison against trixi via 2d advection

### PR DESCRIPTION
**Pull request by pawelmagnu**

### CODE REVIEW

File: examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb
Comment (slayoo) [inline examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: why do we need `assert SETUP["nx"] == SETUP["ny"]`?

Comment (slayoo) [inline examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: we definitely need some introductory narrative in the header explaining:
- what is Trixi
- what is the problem solved and what is the aim of the comparison
- what kind of numerical method and mesh Trixi uses
- what is the approach technically: common JSON file, interpolation using Trixi2VTK, etc
- what and how needs to be installed (e.g., on Colab) to run the notebook

Comment (slayoo) [inline examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: the `def z(t)` function is a copy-paste from the 2D advection-diffusion example, better make reusable there and import here

Comment (slayoo) [inline examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: `omega` (and `mu`, and coordinates min/max) should be part of the shared json

Comment (slayoo) [inline examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: the plot is not visible on GitHub in the notebook

**Review by slayoo — CHANGES_REQUESTED** [review]

File: examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb
Comment (pawelmagnu) [reply examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: the way converting vtu to numpy array works only supports square matrices

Comment (slayoo) [reply examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: let's then comment it with a GitHub issue URL!

Comment (pawelmagnu) [reply examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb:1]: i'll try to fix my code to support rectangular size

**Comment by slayoo** [comment]
@pawelmagnu, it seems that the green "passes" of the notebook execution on Windows and macOS are actually silent failures. Adding some print statements results in no output. On macOS, the following is caught on stderr:
```
_ test_run_notebooks[examples/PyMPDATA_examples/advection_diffusion_2d/advection-diffusion-2d.ipynb] _
----------------------------- Captured stderr call -----------------------------
Assertion failed: false (src/object.cpp:438)
```
It would be best to replace in the notebook the "!" shell magics with Python subprocess execution including exit status checks to know if actually anything was executed and what was the status code.

**Comment by slayoo** [comment]
@pawelmagnu, trying the notebook on Colab (https://colab.research.google.com/github/pawelmagnu/PyMPDATA/blob/trixi_comparison/examples/PyMPDATA_examples/trixi_comparison/advection_comparison.ipynb), it fails due to missing Julia binary:
image

Analogous error will likely happen on mybinder.

Let's try to add a notebook cell with Julia installation magick

**Comment by slayoo** [comment]
@pawelmagnu it works on Colab now!
image
Kudos!

**Review by slayoo — DISMISSED** [review]

**Review by slayoo — APPROVED** [review]

**Pull request merged**